eos_data_distribution/ndn/chunks.py

```diff
@@ -219,6 +219,14 @@ def _on_data(self, o, interest, data):
             self._qualified_name = name.getPrefix(-1)
 
         seg = get_chunk_component(name).toSegment()
+
+        # Have we somehow already got this segment?
+        if self._segments[seg] == SegmentState.COMPLETE:
+            logger.debug('Ignoring data ‘%s’ as it’s already been received',
+                         name)
+            self._check_for_complete()
+            return
```

Comment:
This is a bit of defensive programming. The alternative is to write it as `assert self._segments[seg] != SegmentState.COMPLETE`, if you’re confident enough that NFD/PyNDN will always guarantee this for us. I would suggest the approach in the patch though.